lib/migrator.js

```diff
@@ -6,53 +6,158 @@ const { moveFile, removeDirs } = require('./utils/file')
+
+        // Extract 'components/nested1/nested-component'
+        const classFilePath = filePathFromAppTemplates.slice(1).replace('.hbs', '');
+
+        return !componentsWithPartial.includes(classFilePath);
       });
     }
 
-    sourceComponentTemplateFilePaths.forEach(sourceTemplateFilePath => {
-      let sourceTemplatePathInApp = sourceTemplateFilePath.slice(this.options.projectRoot.length); // '/app/templates/components/nested1/nested-component.hbs'
-      let templatePath = sourceTemplatePathInApp.slice('app/templates/components/'.length); // '/nested1/nested-component.hbs'
-      let targetTemplateFilePath = path.join(this.options.projectRoot, 'app/components', templatePath); // '[APP_PATH]/app/components/nested1/nested-component.hbs'
-      moveFile(sourceTemplateFilePath, targetTemplateFilePath);
-    });
+    return templateFilePaths;
+  }
+
+  changeComponentStructureToFlat(templateFilePaths) {
+    templateFilePaths.forEach(templateFilePath => {
+      // Extract '/app/templates/components/nested1/nested-component.hbs'
+      const filePathFromApp = templateFilePath.slice(this.projectRoot.length);
+
+      // Extract '/nested1/nested-component.hbs'
+      const filePathFromAppTemplatesComponents = filePathFromApp.slice('app/templates/components/'.length);
 
-    templatesWithLayoutName.sort().forEach(template => {
-      console.info(`❌ Did not move '${template}' due to usage as "layoutName" in a component`);
+      // '[APP_PATH]/app/components/nested1/nested-component.hbs'
+      const newTemplateFilePath = path.join(this.projectRoot, 'app/components', filePathFromAppTemplatesComponents);
+      moveFile(templateFilePath, newTemplateFilePath);
     });
-    templatesInPartials.sort().forEach(template => {
-      console.info(`❌ Did not move '${template}' due to usage as a "partial"`);
+  }
+
+  changeComponentStructureToNested(templateFilePaths) {
+    const classFilePaths = this.findClassicComponentClasses();
+
+    templateFilePaths.forEach(templateFilePath => {
+      // Extract '/app/templates/components/nested1/nested-component.hbs'
+      const filePathFromApp = templateFilePath.slice(this.projectRoot.length);
+
+      // Extract '/nested1/nested-component.hbs'
+      const filePathFromAppTemplatesComponents = filePathFromApp.slice('app/templates/components/'.length);
+      const fileExtension = path.extname(filePathFromAppTemplatesComponents);
+
+      // Extract '/nested1/nested-component'
+      const targetPath = filePathFromAppTemplatesComponents.slice(0, -fileExtension.length);
+
+      // Build '[APP_PATH]/app/components/nested1/nested-component/index.hbs'
+      const newTemplateFilePath = path.join(this.projectRoot, 'app/components', targetPath, 'index.hbs');
+      moveFile(templateFilePath, newTemplateFilePath);
+
+      // Build '[APP_PATH]/app/components/nested1/nested-component/index.js'
+      const classFilePath = path.join(this.projectRoot, 'app/components', `${targetPath}.js`);
```

Comment:
Probably should consider `.ts` or `.js`

Reply:
I'll look into this. Thanks for the suggestion!